File: lua/armstrong-numbers/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
	if number == 0 then
		return true
	end

	Sum = 0
	local original_number = number
	local len_number = #tostring(original_number)

	while number > 0 do
		local digit = number % 10
		Sum = Sum + (digit ^ len_number)
		number = math.floor(number / 10)
	end

	return Sum == original_number
end

return ArmstrongNumbers

File: python/armstrong-numbers/armstrong_numbers.py
import math

def is_armstrong_number(number):
    if number == 0: return True
    
    sum = 0
    og_num = number
    len_num = len(str(number))

    while number > 0:
        digit = number % 10
        sum = sum + (digit ** len_num)
        number = math.floor(number / 10)

    return sum == og_num

